src/hooks/useSession.tsx

```diff
@@ -0,0 +1,211 @@
+import React, {
+  createContext,
+  useEffect,
+  useContext,
+  useState,
+  useMemo,
+  useCallback,
+} from 'react';
+import { Account, Subject, Entry, User, SubjectWithProject } from '../types';
+import { useAsyncStorage } from './useAsyncStorage';
+import { useHttpClient } from './useHttpClient';
+import { APIClient } from '@lifeomic/one-query';
+import { RestAPIEndpoints } from '../types/rest-types';
+import { useAuth } from './useAuth';
+
+type Configuration = {
+  account: string;
+  subjects: SubjectWithProject[];
+};
+
+type UserConfiguration = {
+  sessionTime: string;
+  user: User;
+  accounts: Account[];
+  configurations: Configuration[];
+};
+
+type UserConfigurationsContextProps = {
+  userConfiguration: UserConfiguration;
+  isLoaded: boolean;
+  clearSession: () => void;
+};
+
+enum Session {
+  placeholder = 'placeholder',
+}
+
+const configurationPlaceholder: UserConfiguration = {
+  user: {
+    id: Session.placeholder,
+    profile: {
+      email: Session.placeholder,
+    },
+  },
+  configurations: [],
+  accounts: [],
+  sessionTime: new Date().toISOString(),
+};
+
+export const SessionContext = createContext({
+  userConfiguration: configurationPlaceholder,
+  isLoaded: false,
+} as UserConfigurationsContextProps);
+
+export const userSessionKey = 'storedSession';
+
+/**
+ * Uses stored session context to populate infrequently changing values
+ * such as user, account, subjects, projects, and appConfig to improve app loading performance.
+ * A new session is queried for when:
+ * - No existing session is found in storage
+ * - After a session is greater than 7 days old
+ * - An new project invite is accepted
+ * - On logout
+ * - The Clear Session button on the settings screen is pressed (__DEV__ only)
+ */
```

Comment:
Very cool, something we might want to add in the future would be to fetch the app config for the active project in the background and then add a toast that says `there is an app update, tap to update now` or something like that so they get a fast app load but we can avoid them falling too far behind if their cohorts, tiles, or anything has changed that would cause them to get a different app config.

Reply:
That's an excellent idea!